src/WebJobs.Script.WebHost/Diagnostics/WebHostMetricsLogger.cs

```diff
@@ -20,33 +21,71 @@ public WebHostMetricsLogger(IEventGenerator eventGenerator, int metricEventInter
             _metricsEventManager = new MetricsEventManager(eventGenerator, metricEventIntervalInSeconds);
         }
 
+        public object BeginEvent(string eventName)
+        {
+            return _metricsEventManager.BeginEvent(eventName);
+        }
+
         public void BeginEvent(MetricEvent metricEvent)
         {
-            FunctionStartedEvent startedEvent = metricEvent as FunctionStartedEvent;
-            if (startedEvent != null)
+            if (metricEvent is FunctionStartedEvent)
             {
-                startedEvent.StartTime = DateTime.Now;
+                FunctionStartedEvent startedEvent = metricEvent as FunctionStartedEvent;
+                startedEvent.Timestamp = DateTime.Now;
```

Comment:
I'll change to UTC. However currently it doesn't matter because this Timestamp is not logged to the actual event - it's only used for duration calculations.